src/em/prov/em_provisioning.cpp

```diff
@@ -543,6 +545,121 @@ int em_provisioning_t::handle_proxy_encap_dpp(uint8_t *buff, unsigned int len)
+        return nullptr;
+    }
+
+    if (!cJSON_AddItemToArray(b_sta_list_arr, bsta_list_obj)) {
+        printf("%s:%d: Could not add bSTAList object to bSTAList array!\n", __func__, __LINE__);
+        cJSON_Delete(b_sta_list_arr);
+        return nullptr;
+    }
+
+    cJSON *radio_list_arr = cJSON_AddArrayToObject(bsta_list_obj, "RadioList");
+    if (!radio_list_arr) {
+        printf("%s:%d: Could not add RadioList array to bSTAList object!\n", __func__, __LINE__);
+        cJSON_Delete(bsta_list_obj);
+        return nullptr;
+    }
+
+    for (unsigned int i = 0; i < dm->get_num_radios(); i++) {
+        const dm_radio_t *radio = dm->get_radio(i);
+        if (!radio)
+            continue;
+
+        cJSON *radioListObj = cJSON_CreateObject();
+        if (!radioListObj) {
+            printf("%s:%d: Failed to create RadioList object!\n", __func__, __LINE__);
+            cJSON_Delete(bsta_list_obj);
+            return nullptr;
+        }
+
+        if (!cJSON_AddStringToObject(
+                radioListObj, "RUID",
+                util::mac_to_string(radio->m_radio_info.id.ruid, "").c_str())) {
+            printf("%s:%d: Could not add RUID to RadioList object!\n", __func__, __LINE__);
+            cJSON_Delete(bsta_list_obj);
+            return nullptr;
+        }
+
+        std::string radio_channel_list;
+        for (unsigned int j = 0; j < dm->get_num_op_class(); j++) {
+            const dm_op_class_t *opclass = dm->get_op_class(j);
+            if (opclass == nullptr) continue;
+            if (memcmp(opclass->m_op_class_info.id.ruid, radio->m_radio_info.id.ruid, ETH_ALEN) == 0) {
+                radio_channel_list += std::to_string(opclass->m_op_class_info.op_class) + "/" +
+                                      std::to_string(opclass->m_op_class_info.channel);
+                if (j != dm->get_num_op_class() - 1)
+                    radio_channel_list += ",";
+            }
+        }
+        channelList += radio_channel_list;
+
+        if (!cJSON_AddStringToObject(radioListObj, "RadioChannelList",
+                                     radio_channel_list.c_str())) {
+            printf("%s:%d: Could not add RadioChannelList to RadioList object!\n", __func__,
+                   __LINE__);
+            cJSON_Delete(radioListObj);
+            cJSON_Delete(bsta_list_obj);
+            return nullptr;
+        }
+
+        if (!cJSON_AddItemToArray(radio_list_arr, radioListObj)) {
+            printf("%s:%d: Could not add RadioList object to RadioList array!\n", __func__,
+                   __LINE__);
+            cJSON_Delete(radioListObj);
+            cJSON_Delete(bsta_list_obj);
+            return nullptr;
+        }
+    }
+
+    if (!cJSON_AddStringToObject(bsta_list_obj, "channelList", channelList.c_str())) {
+        printf("%s:%d: Could not add channelList to bSTAList object!\n", __func__, __LINE__);
+        cJSON_Delete(bsta_list_obj);
+        return nullptr;
+    }
+    return b_sta_list_arr;
```

Comment:
Isn't the object supposed to be returned here?
```suggestion
    return bsta_list_obj;
```

Reply:
`bSTAList` is the "parent" JSON object here with `bsta_list_obj` inside of it. 

<img width="675" alt="image" src="https://github.com/user-attachments/assets/4abfc02c-dc4b-4edd-8516-0146bc972768" />


`bSTAList` object consists of `akm`, `channelList`, `RadioList` arr, `netRole` and `bSTA_Maximum_Link`

At the callsite in `ec_enrollee.cpp:675` the array is returned and added with `"bSTAList"` as the key to the old DPP Configuration Request Object

Reply:
Got it! I totally missed that. One question. Since this is in `em_t`? Doesn't this only give the info for one BSS? Shouldn't this implementation be located in the `em_agent_t` and passed up via the `em_agent_t` -> `em_mgr_t` (then bound in the `em_t` initializer) similar to `send_action_frame`?